src/components/SingleItem.js

```diff
@@ -1,45 +1,62 @@
 import React, { useEffect } from 'react';
 import { firestore } from '../lib/firebase';
+import calculateEstimate from '../lib/estimates';
 
 const SingleItem = (props) => {
-  const { name, isPurchased, id, token, frequency, lastPurchasedDate } = props;
+  const {
+    name,
+    isPurchased,
+    id,
+    lastPurchasedDate,
+    numberOfPurchases,
+    daysUntilPurchase,
+  } = props;
+
+  const mlsPerDay = 24 * 60 * 60 * 1000;
 
   const updateIsPurchased = async (id) => {
-    await firestore.collection('items').doc(id).set({
+    await firestore.collection('items').doc(id).update({
       isPurchased: false,
-      lastPurchasedDate,
-      token,
-      name,
-      frequency,
     });
   };
 
   setInterval(() => {
     const todaysDate = Date.now();
-    const yesterday = todaysDate - 24 * 60 * 60 * 1000;
-    if (lastPurchasedDate < yesterday) {
+    const yesterday = todaysDate - mlsPerDay;
+    if (lastPurchasedDate && lastPurchasedDate < yesterday) {
       updateIsPurchased(id);
     }
   }, 60000);
 
   useEffect(() => {
     const todaysDate = Date.now();
-    const yesterday = todaysDate - 24 * 60 * 60 * 1000;
-    if (lastPurchasedDate < yesterday) {
+    const yesterday = todaysDate - mlsPerDay;
+    if (lastPurchasedDate && lastPurchasedDate < yesterday) {
       updateIsPurchased(id);
     }
   });
 
+  let latestInterval = 0;
```

Comment:
Based on the equation in estimates.js, should latestInterval be initialized as Number(frequency)? And then leave the following condition as is, so we only update it from the frequency if there is a last purchase date. That way if we have not purchased an item two or more times, there still is an estimated nextPurchaseDate (which is the frequency).

Reply:
@ksiman14 thanks for bringing this up. We were wondering where `frequency` would fit in as well. @segdeha clarified in Slack that `frequency` _is_ one of the inputs for the `calculateEstimate` function. I think it makes sense that we could input that value as the argument for `latestInterval`. I'll test it locally and see how it changes things. @sandaiiyahh 👀

Reply:
Hmm, if we're going by its definition, latestInterval is the number of days between the most recent and the previous purchase. The frequency is more of a user input of how often they would want to buy an item, so I'm not sure 😅. Yeah, let's see how the numbers are like when you make the changes, @isaabutaa.